components/view_image/image_preview.jsx

```diff
@@ -6,9 +6,15 @@ import React from 'react';
 import {getFilePreviewUrl, getFileDownloadUrl} from 'mattermost-redux/utils/file_utils';
 
 export default function ImagePreview({fileInfo, canDownloadFiles}) {
-    const {has_preview_image: hasPreviewImage, id, link} = fileInfo;
-    const fileUrl = link || getFileDownloadUrl(id);
-    const previewUrl = hasPreviewImage ? getFilePreviewUrl(id) : fileUrl;
+    let fileUrl;
+    let previewUrl;
+    if (fileInfo.id) {
```

Comment:
since you have `isExternalFile` on the argument's object, why not use that? it would be more clear
```suggestion
    if (isExternalFile) {
```

Reply:
I meant to remove that actually. I passed both `fileInfo` and `isExternalFile` into the component, but that felt unnecessary, so I tried to remove it. I'll go back and actually remove that, but I'll add `const isExternalFile = Boolean(fileInfo.id)` to make it clearer what's happening